expenses-manager-frontend/src/main/frontend/src/app/expense-table/expense-table.component.html

```diff
@@ -0,0 +1,110 @@
+<div class="datepicker-wrapper">
+  <div class="datepicker-buttons">
+    <button class="date-buttons" (click)="goToPreviousMonth()">&lt;</button>
+    <button class="date-buttons date-picker-button" (click)="openDatepicker(datepicker)">
+      <input matInput [matDatepicker]="datepicker" class="hidden-datepicker-input">
+      {{ displayMonthAndYear() }} <mat-icon>expand_more</mat-icon>
+    </button>
+    <button class="date-buttons" (click)="goToNextMonth()">&gt;</button>
+  </div>
+  <mat-datepicker #datepicker
+                  startView="multi-year"
+                  (monthSelected)="setMonthAndYear($event, datepicker)">
+  </mat-datepicker>
+</div>
+
+<table mat-table [dataSource]="expenses" matSort class="mat-elevation-z8 expense-table">
+  
+  <ng-container matColumnDef="actions">
+    <th mat-header-cell *matHeaderCellDef class="column"> Actions </th>
+    <td mat-cell class="column" *matCellDef="let element"><button class="actions-button"><mat-icon>more_vert</mat-icon></button> </td>
+  </ng-container>
+  
+  <ng-container matColumnDef="date">
+    <th mat-header-cell *matHeaderCellDef mat-sort-header class="column"> Date </th>
+    <td mat-cell class="column" *matCellDef="let element"> {{element.date[2] | number : '2.0-0'}}/{{element.date[1] | number : '2.0-0'}}/{{element.date[0]}} </td>
```

Comment:
Aici mi se pare foarte redundant si greu de citit. Daca vrei sa apici un anumit format pe data iti sugerez sa folosesti un pipe. Vezi aici o documentatie: https://v17.angular.io/api/common/DatePipe

Iar in final codul s-ar scurta: {{ element.date | nume_pipe }}